spring-integration-core/src/test/java/org/springframework/integration/store/SimpleMessageStoreTests.java

```diff
@@ -67,6 +195,110 @@ public void shouldNotHoldMoreThanGroupCapacity() {
+                    try {
+                        Thread.sleep(10);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException("Unexepected",e);
+    
+                    }
+                    store2.removeMessageFromGroup("foo",testMessage2);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    fail("Unexpected exception "+ e.getMessage());
+                }
+                
+            }
+            
+        };
+        store2.addMessageToGroup("foo",testMessage1);
+        t.start();
+        store2.addMessageToGroup("foo",testMessage2);
+        // Wait for both messages to be taken and compared
+        try {
+            t.join();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Unexepected",e);
+
+        }
+    }
+	
+    @Test(expected = MessagingException.class)
+    public void shouldTimeoutAfterWaitIfGroupCapacity() {
+        Thread t = null;
+        try {
+            final SimpleMessageStore store2 = new SimpleMessageStore(1,1,10);
+            final Message<String> testMessage1 = MessageBuilder.withPayload("foo").build();
+            final Message<String> testMessage2 = MessageBuilder.withPayload("bar").build();
+            t = new Thread() {
+    
+                @SuppressWarnings("unchecked")
+                @Override
+                public void run() {
+                    // Simuluate a blocked consumer
+                    try {
+                        Thread.sleep(20);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException("Unexepected",e);
+    
+                    }
+                    store2.removeMessageFromGroup("foo",testMessage1);
+                    // Make sure we yield to allow the second message to be added
+                    try {
+                        Thread.sleep(20);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException("Unexepected",e);
+    
+                    }
+                    fail("Should have timed out in store2.addMessage(testMessage2)");
+                    
+                }
+                
+            };
+            store2.addMessage(testMessage1);
+            t.start();
+            // This should throw
+            store2.addMessage(testMessage2);
+            fail("Should have thrown already");
+            // Wait for both messages to be taken and compared
+        } finally {
+            try {
+                if (t != null) {
+                    t.join();
+                }
+            } catch (InterruptedException e) {
```

Comment:
Please fix any of the style issues as you like. I am not going to be able to fix anything by next week. Just getting my pull request together took a week :)